### ISSUE TITLE
Add sideload support

### ISSUE THREAD

**Issue by nullpixel** [nullpixel/tsssaver#1]
Few ways I can do this:

* Allow manual ECID input on the first attempt, and cache for the rest (easiest)
* Find another private API to get the ECID.

**Comment by nullpixel** [comment]
@AppleBetas your input on this would be liked, since it's evident we cannot get the ECID from inside the box of sand. I could ask the user once and store it somewhere?

**Comment by AppleBetas** [comment]
Not sure if possible but you could:
1. Bundle mobile configuration profile that gets user ECID in app
2. Start temporary server and host profile
3. Redirect user to profile
4. Have URL of profile send data back to app's server, which then redirects it to a URL scheme such as tsssaver://ecid/ECIDHERE

**Comment by AppleBetas** [comment]
I'd say storing the ECID would be convenient, but User Defaults carry over with backups and it's possible that even after switching the device, the user could still get the old device's ECID

**Comment by nullpixel** [comment]
I didn't think that mobileprovs. actually allowed for getting ECIDs. Also, starting a webserver would require the personal VPN entitlement which is paid, so we'd have to host that somewhere in the cloud, which may not be a big issue.

**Comment by AppleBetas** [comment]
You can always just start the server and access it from `DEVICE_INTERNAL_IP_OR_LOCALHOST:3000`.

I think it only has to be from another device when installing apps (what happened with Impactor)

**Comment by nullpixel** [comment]
Ahh, makes sense then. Yeah, I don't see why this'd be an issue, although it could get cumbersome doing it repetitively. How about caching it, and the UDID, and if the UDID has changed then we can re-get the ECID?

**Comment by AppleBetas** [comment]
We can't get the real UDID inside the app without these hoops either, though.

**Comment by nullpixel** [comment]
Ah yeah, that's a valid point too. I'm just not sure from a UX perspective we should be doing this process **every time**. There's no longer any truly unique IDs now, is there?

**Comment by AppleBetas** [comment]
Not that I'm aware of. How often does the user use TSSSaver though? The process is like 30 seconds with this added loophole, so it's not like it is a terrible compromise

**Comment by nullpixel** [comment]
Yeah, that's a good point. I'll add google analytics / fabrics to track users and we can see if it's even worth removing the loophole

**Comment by nullpixel** [comment]
test

**Comment by jakeajames** [comment]
Thanks for the ideas. Gonna make the same thing before you two :P

**Comment by nullpixel** [comment]
No you fucking won't, because I'll license it under ABJJ

**Comment by jakeajames** [comment]
@nullpixel No one needs your code. I am enough smart to make it myself :P

**Comment by nullpixel** [comment]
Your code is fucking disgusting and you don't know what you're doing

**Comment by Sn0wCooder** [comment]
Any news?
